contrib/babelfishpg_common/src/typecode.c

```diff
@@ -300,6 +300,21 @@ lookup_tsql_datatype_oid(const char *typename)
 	return typoid;
 }
 
+Datum
+resolve_pg_type_to_tsql(Oid oid)
+{
+	ht_oid2typecode_entry_t *entry;
+
+	if (OidIsValid(oid))
+	{
+		entry = hash_search(ht_oid2typecode, &oid, HASH_FIND, NULL);
+
+		if (entry && entry->persist_id < TOTAL_TYPECODE_COUNT)
+			PG_RETURN_TEXT_P(CStringGetTextDatum(type_infos[entry->persist_id].tsql_typname));
+	}
+	return (Datum) 0;
```

Comment:
Why are returning datum by converting the cstring and later converting datum to cstring? Why can't we return cstring if we are interested in that only?

Reply:
Changed return type to Void*, no need to convert anymore.